core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -408,4 +431,11 @@ private static String sanitize(char character) {
     }
     return "_x" + Integer.toHexString(character).toUpperCase();
   }
+
+  static boolean isMetricSupportedType(Schema.Type type) {
```

Comment:
How about `supportsMetrics`? I think that would be a bit easier to reason about and it flows in an `if`: `if (supportsMetrics(...)) {...}`